XiEditor/StatusBar.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  StatusBar.swift
+//  XiEditor
+//
+//  Created by Dzũng Lê on 19/05/2018.
+//  Copyright © 2018 Raph Levien. All rights reserved.
+//
+
+import Foundation
+import Cocoa
+
+enum StatusItemAlignment: String {
+    case left = "left"
+    case right = "right"
+}
+
+class StatusItem: NSTextField {
+
+    var key: String = ""
+    var value: String = ""
+    var barAlignment: StatusItemAlignment
+
+    init(_ key: String, _ value: String, _ barAlignment: String) {
+        self.key = key
+        self.value = value
+        self.barAlignment = StatusItemAlignment(rawValue: barAlignment)!
+        super.init(frame: NSZeroRect)
+
+        // Similar to what NSTextField convenience init creates
+        self.isEditable = false
+        self.isSelectable = false
+        self.textColor = NSColor.labelColor
+        self.backgroundColor = NSColor.clear
+        self.lineBreakMode = .byClipping
+        self.isBezeled = false
+        self.stringValue = value
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class StatusBar: NSView {
+
+    private let backgroundColor = NSColor(deviceWhite: 0.9, alpha: 1.0)
```

Comment:
this can probably change based on theme or other settings (you mentioned this, just noting it again)